File: OnlineTaksiJetpack/app/src/main/java/com/example/onlinetaksijetpack/data/remote/model/DriverModels.kt
package com.example.onlinetaksijetpack.data.remote.model

data class DriverLocationUpdateRequest(
    val lat: Double,
    val lng: Double
)

data class DriverProfileResponse(
    val user_id: Int,
    val is_online: Boolean,
    val current_lat: Double?,
    val current_lng: Double?
)

data class AvailableRideItem(
    val id: Int,
    val customer_id: Int,
    val pickup_lat: Double,
    val pickup_lng: Double,
    val pickup_address: String,
    val dropoff_lat: Double,
    val dropoff_lng: Double,
    val dropoff_address: String,
    val status: String,
    val estimated_fare: Double?
)

data class AvailableRideListResponse(
    val rides: List<AvailableRideItem>
)

data class RideListResponse(
    val rides: List<RideResponse>
)

data class DriverOnlineStatusRequest(
    val is_online: Boolean
)

data class UpdateRideStatusRequest(
    val status: String,
    val note: String? = null
)

File: online_taksi_react_native/src/models/driverModels.ts
export type DriverProfileResponse = {
    user_id: number;
    is_online: boolean;
    current_lat?: number | null;
    current_lng?: number | null;
};

export type AvailableRideItem = {
    id: number;
    customer_id: number;
    pickup_lat: number;
    pickup_lng: number;
    pickup_address: string;
    dropoff_lat: number;
    dropoff_lng: number;
    dropoff_address: string;
    status: string;
    estimated_fare?: number | null;
};

export type AvailableRideListResponse = {
    rides: AvailableRideItem[];
};